src/V3Param.cpp

```diff
@@ -806,6 +806,100 @@ class ParamProcessor final {
         }
     }
 
+    void castModuleParamVarValue(const AstNode* nodep, AstVar* modvarp) {
```

Comment:
I was looking at this as have a similar case hitting it.I think instead of this function, the algorithm inside V3Width::visit(AstPin should be pulled into a new V3Width visit that takes the pin and the expression, then performs what would normally be done on a Pin and the pin expression.  V3Width already has I believe all of the cases needed to resolve, and will also handle more complex types.

An added complication is the logic around m_cellp there, please test cell arrays properly de-parameterize.